va/caa_test.go

```diff
@@ -7,10 +7,58 @@ import (
 	"github.com/miekg/dns"
 
 	"github.com/letsencrypt/boulder/core"
+	"github.com/letsencrypt/boulder/features"
 	"github.com/letsencrypt/boulder/probs"
 	"github.com/letsencrypt/boulder/test"
 )
 
+func TestDeepTreeClimb(t *testing.T) {
+	target := "deep-cname.present-with-parameter.com"
```

Comment:
Could you also add a test case that goes `deep-cname.not-present.com -> cname.not-present.com -> not-present.com` so we run through the full branching tree.